release notes/v0.39.0.md

```diff
@@ -0,0 +1,60 @@
+
+We don't expect many scripts to break because of these fixes, but we'll nevertheless announce them as breaking changes in the release notes of the k6 version that they happen in.
+
+## Refactoring metrics, Prometheus remote-write output and tracing support
+
+Over the last several k6 releases, we have also slowly been refactoring and improving the metrics internals in k6 (see [#2071](https://github.com/grafana/k6/pull/2071), [#2330](https://github.com/grafana/k6/issues/2330), [#2426](https://github.com/grafana/k6/pull/2426), [#2463](https://github.com/grafana/k6/pull/2463), [#2433](https://github.com/grafana/k6/pull/2433), [#2442](https://github.com/grafana/k6/pull/2442), among others). This has already brought many side-benefits and minor bugfixes, and we still have a lot of work left (e.g. [#1889](https://github.com/grafana/k6/issues/1889), [#572](https://github.com/grafana/k6/issues/572), [#2430](https://github.com/grafana/k6/issues/2430), [#1831](https://github.com/grafana/k6/issues/1831)), but we have finally reached the point where we can almost start implementing some major new features effectively!
+
+One of the upcoming next big steps is the introduction of a "time series" concept internally in k6 ([#2580](https://github.com/grafana/k6/issues/2580)). We'll start to efficiently group samples (i.e. metric measurements) with the same metric and tags into a single `TimeSeries`, which would unlock many of possibilities that were previously too inefficient to implement. First among them will probably be the refactoring and improvement of the [Prometheus remote-write output extension](https://github.com/grafana/xk6-output-prometheus-remote/) and its eventual merging into k6 as a built-in k6 output.
+
+Another upcoming metrics-related change would be the refactoring of [metric tags](https://k6.io/docs/using-k6/tags-and-groups/#tags) into two distinct types - ones that can be used for indexing (so, in the `TimeSeries` mentioned above, in [sub-metric thresholds](https://k6.io/docs/using-k6/thresholds/#thresholds-on-tags), in certain outputs that do aggregation based on tags, etc. ) and ones that cannot (e.g. high cardinality metadata data that's often unique for every data point). The exact new APIs are still being discussed, but we'll try to make them maximally backwards-compatible. The change is necessary both to implement the time series refactoring mentioned above and also for things like tracing support ([#2128](https://github.com/grafana/k6/issues/2128)).
```

Comment:
https://en.wikipedia.org/wiki/Time_series

@MattDodsonEnglish It's a concept known without the hyphen so I'm not sure we should add it here. Does The Gotham Grammarian still make sense in this case?

Reply:
Since the phrase acts as an adjective (modifying "refactoring"), most style guides, including Gotham Grammarian, would hyphenate here.

That Wikipedia article has a few hyphenated instances (`CTRL+F` `time-series`), usually before "data" or "segmentation": https://en.wikipedia.org/wiki/Time-series_segmentation

But what you say is true: compared to something like the _The New Yorker_, technical documentation is much less likely to hyphenate. Generally, I prefer the hypen because it gives the reader a tiny clue that these words are joined. Sometimes tech docs will have phrases like " packet switched data communication network protocol" which are just awful to read.

But in this case it is a small, small detail, so I certainly won't be sad if it is taken out.

Reply:
I think this change suggestion was for the old version of the release notes, before I split apart the "Refactoring metrics" section? So I couldn't merge it as it was, but I agree with @MattDodsonEnglish that it's easier to read these with hyphenation.